server/webapp/WEB-INF/rails.new/spec/controllers/api_v1/admin/environments_controller_spec.rb

```diff
@@ -0,0 +1,235 @@
+      end
+    end
+  end
+
+  describe :destroy do
+    before(:each) do
+      @environment_name = 'foo-environment'
+      @environment_config = BasicEnvironmentConfig.new(CaseInsensitiveString.new(@environment_name))
+      @environment_config_service = double('environment-config-service')
+      controller.stub(:environment_config_service).and_return(@environment_config_service)
+      @environment_config_service.stub(:getEnvironmentConfig).with(@environment_name).and_return(@environment_config)
+    end
+
+    describe :for_admins do
+      it 'should allow deleting environments' do
+        login_as_admin
+
+        @environment_config_service.should_receive(:deleteEnvironment).with(@environment_name,an_instance_of(Username), an_instance_of(HttpLocalizedOperationResult)) do |name, user, result|
+          result.setMessage(LocalizedMessage.string('ENVIRONMENT_DELETE_SUCCESSFUL', [name].to_java(java.lang.String)))
+        end
+        controller.request.env['HTTP_IF_MATCH'] = "\"#{Digest::MD5.hexdigest('latest-etag')}\""
+
+        delete_with_api_header :destroy, name: @environment_name
+        expect(response.status).to eq(200)
+        expect(response).to have_api_message_response(200, "Environment 'foo-environment' was deleted successfully.")
+      end
+
+      it 'should render 404 when a environment does not exist' do
+        login_as_admin
+
+        @environment_name = SecureRandom.hex
+        @environment_config_service.stub(:getEnvironmentConfig).and_raise(com.thoughtworks.go.config.exceptions.NoSuchEnvironmentException.new('foo-env'))
+        delete_with_api_header :destroy, name: @environment_name
+        expect(response).to have_api_message_response(404, 'Either the resource you requested was not found, or you are not authorized to perform this action.')
+      end
+    end
+
+  end
+
+  describe :create do
+    before(:each) do
+      @environment_name = 'foo-environment'
+      @environment_config = BasicEnvironmentConfig.new(CaseInsensitiveString.new(@environment_name))
+      @environment_config_service = double('environment-config-service')
+      controller.stub(:environment_config_service).and_return(@environment_config_service)
+      @environment_config_service.stub(:getEnvironmentConfig).with(@environment_name).and_return(@environment_config)
+    end
+
+
+    describe :for_admins do
+      it 'should render 200 created when environment is created' do
+        login_as_admin
+
+        @environment_config_service.should_receive(:createEnvironment)
+
+        post_with_api_header :create, :environment => { name: @environment_name, pipelines: [], agents: [], environment_variables: []}
+        expect(response.status).to be(200)
+        expect(actual_response).to eq(expected_response(@environment_config, ApiV1::Config::EnvironmentConfigRepresenter))
+      end
+    end
+
```

Comment:
Perhaps there are some scenarios when environment cannot be created? 
Example: If environment name already exists